zkevm-circuits/src/copy_circuit.rs

```diff
@@ -265,25 +280,31 @@ impl<F: Field> SubCircuitConfig<F> for CopyCircuitConfig<F> {
                     );
                 },
             );
+            cb.require_equal(
+                "write value == read value",
+                meta.query_advice(value, Rotation::cur()),
+                meta.query_advice(value, Rotation::next()),
+            );
+            cb.require_equal(
+                "value_acc_rlc is same for read-write rows",
+                meta.query_advice(value_acc_rlc, Rotation::cur()),
+                meta.query_advice(value_acc_rlc, Rotation::next()),
+            );
             cb.condition(
-                not::expr(tag.value_equals(CopyDataType::RlcAcc, Rotation::next())(
-                    meta,
-                )),
+                and::expr([
+                    not::expr(meta.query_advice(is_last, Rotation::next())),
+                    not::expr(meta.query_advice(is_pad, Rotation::cur())),
+                ]),
                 |cb| {
                     cb.require_equal(
-                        "write value == read value (if not rlc acc)",
-                        meta.query_advice(value, Rotation::cur()),
-                        meta.query_advice(value, Rotation::next()),
+                        "value_acc_rlc(2) == value_acc_rlc(0) * r + value(2)",
+                        meta.query_advice(value_acc_rlc, Rotation(2)),
+                        meta.query_advice(value_acc_rlc, Rotation::cur())
+                            * challenges.keccak_input()
+                            + meta.query_advice(value, Rotation(2)),
```

Comment:
It seems we don't have constraint to enforce for the first step of each copy event, the `value_acc_rlc` should be equal to `value`. A malicious prover now can start with any value.

Reply:
You're right. After changing the layout I must have forgotten to re-add that constraint. Thank you for noticing!

Reply:
fixed in [d2bdc68](https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1419/commits/d2bdc68414d218732a56d4e1d038b1c11859c9e6)

Reply:
The following question is, is there any constraint to enforce `is_first` to be `1` when we are entering next copy event? Since I couldn't find an explicit one for that, and after I change the `is_first` assignment to all `0` the tests are still passing.

Reply:
https://github.com/privacy-scaling-explorations/zkevm-circuits/issues/1432 created for `is_first` and `is_last`